backstop/src/backstop/user.rs

```diff
@@ -117,14 +113,14 @@ impl UserBalance {
     /// Withdraw shares from the user
     ///
     /// ### Arguments
-    /// * `to_q` - The amount of new shares to queue for withdraw
+    /// * `to_dequeue` - The amount of shares to dequeue from withdraw
     ///
     /// ### Errors
     /// If the amount to queue is greater than the available shares
-    pub fn withdraw_shares(&mut self, e: &Env, to_withdraw: i128) {
-        self.dequeue_shares_for_withdrawal(e, to_withdraw, true);
+    pub fn dequeue_withdrawal(&mut self, e: &Env, to_dequeue: i128) {
+        self.dequeue_shares_for_withdrawal(e, to_dequeue, false);
 
-        self.shares -= to_withdraw;
+        self.shares += to_dequeue;
     }
```

Comment:
Recommending that this function be deleted.

This functionality should be orchestrated in `execute_dequeue_withdrawal`, as it would roughly go:

```
/// Perform a dequeue of queued for withdraw deposits from the backstop module
pub fn execute_dequeue_withdrawal(e: &Env, from: &Address, pool_address: &Address, amount: i128) {
    require_nonnegative(e, amount);
    let mut pool_balance = storage::get_pool_balance(e, pool_address);
    let mut user_balance = storage::get_user_balance(e, pool_address, from);

    user_balance.dequeue_shares_for_withdrawal(e, amount, false);
    // update emissions
    emissions::update_emissions(e, pool_address, &pool_balance, from, &user_balance, false);
    
// dequeued shares from the withdrawal queue and return them to the user
    user_balance. dequeue_shares_for_withdrawal(e, amount, false);
user_balance.add_shares(amount);
    pool_balance.dequeue_q4w(e, amount);

    storage::set_user_balance(e, pool_address, from, &user_balance);
    storage::set_pool_balance(e, pool_address, &pool_balance);
}
```